### ISSUE TITLE
After update, keeps saying a newer verion is available

### ISSUE THREAD

**Issue by ebogaard** [pymedusa/Medusa#9895]
**Describe the bug**
Since the last few upgrades, Medusa doesn't seem to know it's upgraaded and keeps showing notifications like "There is a newer version available (you're 20 commits behind) — Update Now".
If I compare the files to the ones on Github, I can see they are the correct versions.

**To Reproduce**
1. Follow the "Update now" process
2. Acknoledge the update request
3. Acknowledge restarting the service when asked.
4. To be sure, restart the service via systemd as well

This process has worked for months, until the update about a month (maybe two) ago.

**Expected behavior**
After following this procedure, I expect the "Newer version available"-message to disappear. For some time that is actually the case, until I manually check for new versions or that process is automatically triggered in the background after a while.

**Medusa (please complete the following information):**
 - OS: AlmaLinux 8.4
 - Branch: master
 - Commit: 
 - Python version: 3.6.8
 - Database version: ?

**Debug logs (at least 50 lines):**
General > Advanced Settings > Enable debug
<details>

```
2021-09-20 15:06:45 DEBUG    APIv2-Thread_0 :: [25e7b60] cur_commit = 25e7b6072201a8a73e1220ee5ffe13cbefce45f8, newest_commit = 21e350cd4a21f6fb65f750ab5c2d1e9c92c621d6, num_commits_behind = 20
2021-09-20 15:06:45 INFO     APIv2-Thread_4 :: [25e7b60] Config backup in progress...
2021-09-20 15:06:48 INFO     APIv2-Thread_4 :: [25e7b60] Config backup successful
2021-09-20 15:06:48 INFO     APIv2-Thread_3 :: [25e7b60] Clearing update folder '/home/**********/Medusa/medusa-update' before extracting
2021-09-20 15:06:48 INFO     APIv2-Thread_3 :: [25e7b60] Downloading update from 'http://github.com/pymedusa/Medusa/tarball/master'
2021-09-20 15:06:48 DEBUG    APIv2-Thread_3 :: [25e7b60] GET URL: http://github.com/pymedusa/Medusa/tarball/master [Status: 301]
2021-09-20 15:06:48 DEBUG    APIv2-Thread_3 :: [25e7b60] User-Agent: Medusa/0.5.18 (Linux; 4.18.0-305.17.1.el8_4.x86_64; 3ab2ab66-1599-11ec-8b6b-bc5ff4f2e894)
2021-09-20 15:06:48 DEBUG    APIv2-Thread_3 :: [25e7b60] GET URL: https://github.com/pymedusa/Medusa/tarball/master [Status: 302]
2021-09-20 15:06:48 DEBUG    APIv2-Thread_3 :: [25e7b60] User-Agent: Medusa/0.5.18 (Linux; 4.18.0-305.17.1.el8_4.x86_64; 3ab2ab66-1599-11ec-8b6b-bc5ff4f2e894)
2021-09-20 15:06:49 DEBUG    APIv2-Thread_3 :: [25e7b60] GET URL: https://codeload.github.com/pymedusa/Medusa/legacy.tar.gz/master [Status: 200]
2021-09-20 15:06:49 DEBUG    APIv2-Thread_3 :: [25e7b60] User-Agent: Medusa/0.5.18 (Linux; 4.18.0-305.17.1.el8_4.x86_64; 3ab2ab66-1599-11ec-8b6b-bc5ff4f2e894)
2021-09-20 15:06:50 INFO     APIv2-Thread_3 :: [25e7b60] Extracting file /home/**********/Medusa/medusa-update/medusa-update.tar
2021-09-20 15:06:53 INFO     APIv2-Thread_3 :: [25e7b60] Deleting file /home/**********/Medusa/medusa-update/medusa-update.tar
2021-09-20 15:06:53 INFO     APIv2-Thread_3 :: [25e7b60] Moving files from /home/**********/Medusa/medusa-update/pymedusa-Medusa-21e350c to /home/**********/Medusa
2021-09-20 15:06:53 DEBUG    APIv2-Thread_3 :: [25e7b60] Special handling for MediaInfo.dll
2021-09-20 15:06:53 DEBUG    APIv2-Thread_3 :: [25e7b60] Special handling for MediaInfo.dll
2021-09-20 15:06:53 DEBUG    APIv2-Thread_3 :: [] Notification for Boxcar2 not enabled, skipping this notification
2021-09-20 15:08:24 INFO     EVENT-QUEUE :: [] Aborting all threads
2021-09-20 15:08:24 INFO     EVENT-QUEUE :: [] Restarting Medusa with ['/usr/bin/python3', '/home/**********/Medusa/start.py', '-q', '--nolaunch', '--datadir=/home/**********/Medusa']


```
</details>

**Additional context**

**Comment by medariox** [comment]
The current release is a few commits behind master, that's probably the reason why you get the message. I highly suggest to use GitHub or even better to use the installer we provide.

**Comment by nydave69** [comment]
I see the same thing...   I am now 6 commits behind according to the banner..
Also when I click the banner to see the update info, I get a screen asking me to pick the versions I want to compare.

**Comment by ebogaard** [comment]
> The current release is a few commits behind master, that's probably the reason why you get the message. I highly suggest to use GitHub or Docker.

So you're saying the built-in updater doesn't actually update to the latest commit? What is it updating to then?
After updating, I can see that the files in the Medusa folder on my server are the actual verions from master (for example: CHANGELOG.md contains notes about 0.5.18), so to me it seems they are current (21e350cd4a21f6fb65f750ab5c2d1e9c92c621d6) and not from the old commit Medusa thinks it's still on (25e7b6072201a8a73e1220ee5ffe13cbefce45f8).

I can't find any active branches that are exactly 20 commits ahead of master, so I can't explain this message.

If I do a 'manual' "Checkout branch" via Settings > General > Advanced > Github, and have branch version = master selected, I see the following in the logs:
```
2021-09-21 14:05:50 DEBUG    Thread_0 :: [25e7b60] cur_commit = 25e7b6072201a8a73e1220ee5ffe13cbefce45f8, newest_commit = 21e350cd4a21f6fb65f750ab5c2d1e9c92c621d6, num_commits_behind = 20
2021-09-21 14:05:50 DEBUG    Thread_0 :: [25e7b60] GET URL: http://rawcdn.githack.com/pymedusa/Medusa/21e350cd4a21f6fb65f750ab5c2d1e9c92c621d6/medusa/databases/main_db.py [Status: 200]
2021-09-21 14:05:50 DEBUG    Thread_0 :: [25e7b60] User-Agent: Medusa/0.5.18 (Linux; 4.18.0-305.17.1.el8_4.x86_64; d7670686-1a13-11ec-835f-bc5ff4f2e894)
2021-09-21 14:05:50 DEBUG    Thread_0 :: [25e7b60] Checkout branch has the same DB version - Equal
```

Could it be that the DB isn't updated to reflect the newly checked out version?

**Comment by c0dewhacker** [comment]
I Have this exact same issue. using the pymedusa/medusa:latest image tag to pull the docker image.

e.g. latest version 0.5.29
but still 300 commits behind.
![image](https://user-images.githubusercontent.com/27728386/170849408-f91e12cc-963b-4d31-9faf-ce4a17a81fb3.png)

This also means i get an error when trying to submit an issue directly from the app because i'm not on the latest version. 
![image](https://user-images.githubusercontent.com/27728386/170849461-ff4b9f5c-dea2-478e-9592-c351f1877628.png)

I seem to be (feature wise) on the latest version. e.g. i can exclude cache.db from backups etc... but the app doesn't appear to think it is on the latest version.

**Comment by c0dewhacker** [comment]
@nydave69 @ebogaard @medariox 

It appears that this issue is resolved with v1.0.2

No longer getting any " X commits behind" banner after pulling the latest docker image.

**Comment by medariox** [comment]
Great to hear! Closing as solved.
